### PR TITLE
[Merged by Bors] - chore: update Mathlib/Tactic.lean

### PR DESCRIPTION
**Pull request by kbuzzard**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

Tactics like `propose` are now included. I also sorted the file into alphabetical order, and added files in subdirectories, which explains part of the diff.

On the way I noticed that there are a few examples of directories without corresponding files, e.g. there is `Tactic/Relation/` but no `Tactic/Relation.lean`. Presumably we don't care about this.

I'll also remark that I'm preparing these files in a very clunky way, doing `find . -iname "*"` and then manually editing the output in a text editor with search and replace etc. It's still very much worth doing as far as I'm concerned though -- I've done this one because I was annoyed that `import Mathlib.Tactic` didn't get me `propose`.

### CODE REVIEW

File: Mathlib/Tactic.lean [hunk 1]
```diff
@@ -44,33 +48,60 @@ import Mathlib.Tactic.LeftRight
 import Mathlib.Tactic.LibrarySearch
 import Mathlib.Tactic.Lift
 import Mathlib.Tactic.Linarith
+import Mathlib.Tactic.Linarith.Datatypes
+import Mathlib.Tactic.Linarith.Elimination
+import Mathlib.Tactic.Linarith.Frontend
+import Mathlib.Tactic.Linarith.Lemmas
+import Mathlib.Tactic.Linarith.Parsing
+import Mathlib.Tactic.Linarith.Preprocessing
+import Mathlib.Tactic.Linarith.Verification
```
Comment (semorrison) [inline Mathlib/Tactic.lean:57]: This is harmless, but they are all imported via `Mathlib.Tactic.Linarith`.

**Comment by semorrison** [comment]
Would it be better to just add the same CI that manages `Mathlib.lean`, and just require everything in `Mathlib/Tactic/` is imported in `Mathlib/Tactic.lean`?

**Comment by kbuzzard** [comment]
> Would it be better to just add the same CI that manages `Mathlib.lean`, and just require everything in `Mathlib/Tactic/` is imported in `Mathlib/Tactic.lean`?

I personally would be very happy with this solution but would not be able to make that CI change myself.

**Comment by semorrison** [comment]
I added the CI step to this PR.

(The process for updating CI is to edit `.github/workflows/build.yml.in`, and then run `.github/workflows/mk_build_yml.sh` to auto-generate the other files.)

**Comment by semorrison** [comment]
That failed, because the script has a different idea about alphabetical order. :-)

**Comment by semorrison** [comment]
bors d+

**Pull request closed without merging**